common/scala/src/main/scala/whisk/connector/kafka/KafkaConsumerConnector.scala

```diff
@@ -69,13 +70,18 @@ class KafkaConsumerConnector(
         val thread = new Thread() {
             override def run() = {
                 while (!disconnect) {
-                    Try { peek() } match {
-                        case Success(records) =>
-                            records foreach { process.tupled(_) }
-                        case Failure(t) =>
-                            error(self, s"while polling: $t")
+                    Try {
+                        // Grab next batch of messages and commit offsets immediately
+                        // If the commit fails, the message will not be processed and fetched again in the next run
+                        val records = peek()
+                        commit()
+                        records
+                    } map { records =>
+                        records foreach { process.tupled(_) }
```

Comment:
put on one lines or use `_`? I think it's pretty clear the code is processing records.